### PR TITLE
[BACKLOG-35909] added new class and changed code to use new class

### PR DESCRIPTION
**Pull request by MarcDiamantHitachi**

### CODE REVIEW

**Comment by buildguy** [comment]
### :warning: Build finished in 36m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.TransTest.testPDI12424ParametersFromMetaAreCopiedToTrans</b> (click to expand)</summary><p>

```

Transformation was unable to open [null]

The transformation path ram:///parameters_4df17d46-903d-11ed-97f0-45c6c4293a4b.ktr is invalid, and will not run successfully.


```

</p></details>

  
Tests run: **4030**, Failures: **1**, Skipped: **8** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi-11/job/PR-8772/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/main/java/org/pentaho/di/trans/steps/delete/Delete.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public class Delete extends BaseStep implements StepInterface {
   private DeleteData data;
 
   public Delete( StepMeta stepMeta, StepDataInterface stepDataInterface, int copyNr, TransMeta transMeta,
-    Trans trans ) {
+                 Trans trans ) {
```
Comment (peterrinehart) [inline engine/src/main/java/org/pentaho/di/trans/steps/delete/Delete.java:57]: Not sure why this formatting needed to change; this doesn't look aligned with our code standards.

File: engine/src/main/java/org/pentaho/di/trans/steps/delete/Delete.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ private synchronized void deleteValues( RowMetaInterface rowMeta, Object[] row )
 
     if ( log.isDebug() ) {
       logDebug( BaseMessages.getString( PKG, "Delete.Log.SetValuesForDelete", data.deleteParameterRowMeta
-        .getString( deleteRow ), rowMeta.getString( row ) ) );
+              .getString( deleteRow ), rowMeta.getString( row ) ) );
```
Comment (peterrinehart) [inline engine/src/main/java/org/pentaho/di/trans/steps/delete/Delete.java:82]: Check indent spacing; standard is two spaces per level of indent.

File: engine/src/main/java/org/pentaho/di/trans/steps/delete/DeleteMeta.java [hunk 3]
```diff
@@ -305,23 +240,24 @@ private void readData( Node stepnode, List<? extends SharedObjectInterface> data
...
     } catch ( Exception e ) {
       throw new KettleXMLException( BaseMessages.getString(
-        PKG, "DeleteMeta.Exception.UnableToReadStepInfoFromXML" ), e );
+              PKG, "DeleteMeta.Exception.UnableToReadStepInfoFromXML" ), e );
     }
   }
 
   public void setDefault() {
-    keyStream = null;
+    //keyStream = null;
```
Comment (peterrinehart) [inline engine/src/main/java/org/pentaho/di/trans/steps/delete/DeleteMeta.java:259]: Remove this commented out line

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: engine/src/main/java/org/pentaho/di/trans/steps/delete/DeleteMeta.java [hunk 4]
```diff
@@ -336,19 +272,19 @@ public String getXML() {
     StringBuilder retval = new StringBuilder( 500 );
 
     retval
-      .append( TAG_4_SPACES ).append(
-        XMLHandler.addTagValue( TAG_CONNECTION, databaseMeta == null ? "" : databaseMeta.getName() ) );
+            .append( TAG_4_SPACES ).append(
```
Comment (peterrinehart) [inline engine/src/main/java/org/pentaho/di/trans/steps/delete/DeleteMeta.java:275]: Check indent levels

**Pull request merged**